### ISSUE TITLE
Default value for session_timeout_ms user resource

### ISSUE THREAD

**Issue by milosz** [suzuki-shunsuke/go-graylog#304]
## Note

I think that there should be default value for session_timeout_ms user resource or this attribute should be required.

## Environment

* Terraform version: Terraform v0.11.11 (snap)
* Graylog version: Graylog 3.3.4+279ef2c on 99c4d56e8a80 (Oracle Corporation 1.8.0_265 on Linux 5.4.0-42-generic)
* terraform-provider-graylog or go-graylog version: terraform-provider-graylog_1.0.4_linux_amd64.zip (latest)
* OS (client side): Ubuntu 20.04

## Overview

There should be default value for session_timeout_ms or this attribute should be required.

Create user.

```
provider "graylog" {                                                                                                                      
  web_endpoint_uri = "https://graylog.example.com/api"                                                                   
  api_version      = "v3"                                                                                                                 
  auth_name        = "admin"                                                                                                              
  auth_password    = "admin"                                                                                                           
}                                                                                                                                         
                                                                                                                                                                                                                                                                                   
resource "graylog_user" "graylog" {                                                                                                       
  username  = "graylog"                                                                                                                   
  email     = "test@example.com"                                                                                                          
  full_name = "graylog"                                                                                                                   
  password  = "graylog"                                                                                                                   
  roles = [                                                                                                                               
    "Reader"                                                                                                                              
  ]                                                                                                                                       
}      
```

```
$ terraform plan -out plan
Refreshing Terraform state in-memory prior to plan...                                                                                     
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.
                                                                     
graylog_user.graylog: Refreshing state... (ID: graylog)                                                                                   
                                                                     
------------------------------------------------------------------------
                                                                     
An execution plan has been generated and is shown below.             
Resource actions are indicated with the following symbols: 
  + create                                                           
                                                                     
Terraform will perform the following actions:           
                                                                                                                                          
  + graylog_user.graylog                                             
      id:                 <computed>                                                                                                      
      client_address:     <computed>       
      email:              "test@example.com"                         
      external:           <computed>                   
      full_name:          "graylog"                                  
      last_activity:      <computed>                                                                                                      
      password:           <sensitive>                                
      permissions.#:      <computed>                    
      read_only:          <computed>                       
      roles.#:            "1"                                        
      roles.2199354427:   "Reader"                                                                                                        
      session_active:     <computed>
      session_timeout_ms: <computed>
      timezone:           <computed>
      user_id:            <computed>
      username:           "graylog"


Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

This plan was saved to: plan

To perform exactly these actions, run the following command to apply: 
    terraform apply "plan"
```
```
$ terraform apply plan                                                
graylog_user.graylog: Creating...
  client_address:     "" => "<computed>"
  email:              "" => "test@example.com"
  external:           "" => "<computed>"
  full_name:          "" => "graylog"
  last_activity:      "" => "<computed>"
  password:           "<sensitive>" => "<sensitive>"
  permissions.#:      "" => "<computed>"
  read_only:          "" => "<computed>"
  roles.#:            "" => "1"
  roles.2199354427:   "" => "Reader"
  session_active:     "" => "<computed>"
  session_timeout_ms: "" => "<computed>"
  timezone:           "" => "<computed>"
  user_id:            "" => "<computed>"
  username:           "" => "graylog"
graylog_user.graylog: Creation complete after 0s (ID: graylog)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
I cannot login using created user.

## Expected behavior

Login successfully to graylog as created user.

## Actual behavior

I cannot login as created user, graylog log shows the culprit:

```
graylog_graylog.1.xbjfwq7oku9d@swarm    | 2020-08-19 20:56:25,317 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
graylog_graylog.1.xbjfwq7oku9d@swarm    | org.apache.shiro.session.ExpiredSessionException: Session with id [fd8526df-e33c-48e1-a2b5-358b9e4e067c] has expired. Last access time: 8/19/20 8:56 PM.  Current time: 8/19/20 8:56 PM.  Session timeout is set to 0 seconds (0 minutes)
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.SimpleSession.validate(SimpleSession.java:292) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.AbstractValidatingSessionManager.doValidate(AbstractValidatingSessionManager.java:186) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.AbstractValidatingSessionManager.validate(AbstractValidatingSessionManager.java:143) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.AbstractValidatingSessionManager.doGetSession(AbstractValidatingSessionManager.java:120) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.AbstractNativeSessionManager.lookupSession(AbstractNativeSessionManager.java:148) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.AbstractNativeSessionManager.lookupRequiredSession(AbstractNativeSessionManager.java:152) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.AbstractNativeSessionManager.touch(AbstractNativeSessionManager.java:231) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.mgt.DelegatingSession.touch(DelegatingSession.java:120) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.apache.shiro.session.ProxiedSession.touch(ProxiedSession.java:100) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.graylog2.shared.security.SessionCreator.create(SessionCreator.java:89) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.graylog2.rest.resources.system.SessionsResource.newSession(SessionsResource.java:127) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_265]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_265]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_265]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_265]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) ~[graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:181) [graylog.jar:?]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_265]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_265]
graylog_graylog.1.xbjfwq7oku9d@swarm    |        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
                                                                 
```

## Fix

It works when I define session_timeout_ms and apply plan.

```
provider "graylog" {                                                                                                                      
  web_endpoint_uri = "https://graylog.example.com/api"                                                                   
  api_version      = "v3"                                                                                                                 
  auth_name        = "admin"                                                                                                              
  auth_password    = "admin"                                                                                                           
}                                                                                                                                         
                                                                                                                                                                                                                                                                                   
resource "graylog_user" "graylog" {                                                                                                       
  username  = "graylog"                                                                                                                   
  email     = "test@example.com"                                                                                                          
  full_name = "graylog"                                                                                                                   
  password  = "graylog"                                                                                                                   
  session_timeout_ms = "3600000"                                                                                                          
  roles = [                                                                                                                               
    "Reader"                                                                                                                              
  ]                                                                                                                                       
}      
```
```
$ terraform plan -out plan                                                                                             
Refreshing Terraform state in-memory prior to plan...                                                                                      The refreshed state will be used to calculate this plan, but will not be                                                                   persisted to local or remote state storage.                                                                                                                                                                                                                                           graylog_user.graylog: Refreshing state... (ID: graylog)                                                                                                                                                                                                                               ------------------------------------------------------------------------                                                                                                                                                                                                              An execution plan has been generated and is shown below.                                                                                   Resource actions are indicated with the following symbols:                                                                                   + create                                                                                                           
                                                                                                                                           Terraform will perform the following actions:                                                                                                                                                                                                                                           + graylog_user.graylog                                                                                                                         id:                 <computed>                                                                                                             client_address:     <computed>                                                                                                             email:              "test@example.com"                                                                                                     external:           <computed>                                                                                                             full_name:          "graylog"                                                                                                              last_activity:      <computed>                                                                                                             password:           <sensitive>                                                                                                            permissions.#:      <computed>                                
      read_only:          <computed>                                                      
      roles.#:            "1"
      roles.2199354427:   "Reader" 
      session_active:     <computed>
      session_timeout_ms: "3600000"
      timezone:           <computed>
      user_id:            <computed>
      username:           "graylog"


Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------                                                                   

This plan was saved to: plan

To perform exactly these actions, run the following command to apply:                                                                      
    terraform apply "plan"
```

```
$ terraform apply plan                                                                                                                     
graylog_user.graylog: Creating...
  client_address:     "" => "<computed>"
  email:              "" => "test@example.com"
  external:           "" => "<computed>"
  full_name:          "" => "graylog"
  last_activity:      "" => "<computed>"
  password:           "<sensitive>" => "<sensitive>"
  permissions.#:      "" => "<computed>"
  read_only:          "" => "<computed>"
  roles.#:            "" => "1"
  roles.2199354427:   "" => "Reader"
  session_active:     "" => "<computed>"
  session_timeout_ms: "" => "3600000"
  timezone:           "" => "<computed>"
  user_id:            "" => "<computed>"
  username:           "" => "graylog"
graylog_user.graylog: Creation complete after 0s (ID: graylog)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
I can login at this moment.

**Comment by suzuki-shunsuke** [comment]
Thank you for your report.

I have confirmed the behavior on Graylog 3.2.4+a407287.

When we create a user from Web UI, the default value of the timeout is 1 hour (3600000).

![image](https://user-images.githubusercontent.com/13323303/90696216-d2fde000-e2b6-11ea-80e4-c0c3d345e417.png)

When we check `Sessions do not time out`, the value of the `session_timeout_ms` becomes `-1`.

**Comment by suzuki-shunsuke** [comment]
We'll set the default value `3600000` and change `Computed` to `false`.
https://github.com/suzuki-shunsuke/go-graylog/pull/305

**Comment by milosz** [comment]
Thanks

**Comment by suzuki-shunsuke** [comment]
I have released [v12.0.0](https://github.com/suzuki-shunsuke/go-graylog/releases/tag/v12.0.0)
